interfaces/repo.go

```diff
@@ -322,68 +322,100 @@ func (r *Repository) Connect(plugSnapName, plugName, slotSnapName, slotName stri
 //   from all the slots found therein. It is not an error if there are no
 //   such plugs but it is still an error if the snap does not exist or has no
 //   slots at all.
-func (r *Repository) Disconnect(plugSnapName, plugName, slotSnapName, slotName string) error {
+//
+// Disconnect returns the list of severed connections, the list of affected snaps
+// and an error if one occurred.
+func (r *Repository) Disconnect(plugSnapName, plugName, slotSnapName, slotName string) ([]ConnRef, []*snap.Info, error) {
```

Comment:
@niemeyer yes, we had an issue like that. I think it might be avoided. In my earlier attempt at this branch I did it differently and instead returned a list of connections that were severed. This didn't contain the snap info but did contain enough data to do the right thing in the overlord. I can change it to behave like that